packages/jest-haste-map/src/index.js

```diff
@@ -325,6 +325,28 @@ class HasteMap extends EventEmitter {
           throw new Error(message);
         }
         this._console.warn(message);
```

Comment:
I disagree, we need to leave it so that people know there's duplication during watchmode. Without it it would say "module doesn't exist", without any actionable feedback.

What we can do later is improve the API so that consumers can identify duplicated IDs.